tools/cldr-code/src/test/java/org/unicode/cldr/unittest/TestPersonNameFormatter.java

```diff
@@ -81,21 +81,21 @@ public void TestBasic() {
 
         NamePatternData namePatternData = new NamePatternData(
             localeToOrder,
-            "length=short; style=formal; usage=addressing; order=surnameFirst", "{surname-allCaps} {given}",
-            "length=short medium; style=formal; usage=addressing", "{given} {given2-initial} {surname}",
-            "length=short medium; style=formal; usage=addressing", "{given} {surname}",
-            "length=long; style=formal; usage=monogram", "{given-initial}{surname-initial}",
+            "order=surnameFirst; length=short; usage=addressing; formality=formal", "{surname-allCaps} {given}",
+            "length=short medium; usage=addressing; formality=formal", "{given} {given2-initial} {surname}",
+          //"length=short medium; usage=addressing; formality=formal", "{given} {surname}", // TODO: Got error "key is masked by previous values" for this; OK?
```

Comment:
Without commenting, got error "key is masked by previous values". Not sure what the expectation is here since the parameters duplicate the previous entry.

Reply:
Some progress. The problem is in

public NamePatternData(ImmutableMap<ULocale, Order> localeToOrder,
            ImmutableListMultimap<ParameterMatcher, NamePattern> formatParametersToNamePattern) {

In formatParametersToNamePattern.asMap() there should never be two duplicate keys. I think something has gone wrong in the equals() comparison, and will track down.

Reply:
The problem was in
            return orders.equals(that.orders)
                && lengths.equals(that.usages)
                && usages.equals(that.lengths)
                && formalities.equals(that.formalities);

somehow the lengths are being compared to usages.

Reply:
Fixing in my local copy and going to next one.

Reply:
Oops, that was my fault, thanks for tracking down!!